client/my-sites/promote-post/main.tsx

```diff
@@ -81,6 +94,21 @@ export default function PromotedPosts( { tab }: Props ) {
 		isRequestingPostsForQuery( state, selectedSiteId, queryProducts )
 	);
 
+	const topViewedPostAndPages = useSelector( ( state ) =>
+		getTopPostAndPages( state, selectedSiteId, topPostsQuery )
+	);
+
+	const mostPopularPostAndPages = useSelector( ( state ) => {
+		const topPosts: any[ PostType ] = [];
+
+		topViewedPostAndPages?.map( ( post: any ) => {
+			const item = getSitePost( state, selectedSiteId, post.id );
+			item && topPosts.push( { ...item, views: post.views } );
+		} );
+
+		return topPosts;
```

Comment:
Should we also return the pages?

Reply:
Good point, will add the parameter to return `any` type of posts.

Reply:
Noted it also returns pages, so [I just renamed the variable here](https://github.com/Automattic/wp-calypso/pull/73760/commits/905023029e6a132ce88d9a7290707a23fa6334f5).